File: Backend/src/main/java/projectinsight/module/project/domain/employee/model/Employee.java
package projectinsight.module.project.domain.employee.model;


import java.time.Instant;

public class Employee {

  protected String id;
  protected String name;
  protected String surname;
  protected Instant creationInstant;
  protected Instant lastUpdateInstant;
  protected boolean deleted;

  public String getId() {
    return id;
  }
  public String getName() {
    return name;
  }
  public String getSurname() {
    return surname;
  }
  public Instant getCreationInstant() {
    return creationInstant;
  }
  public Instant getLastUpdateInstant() {
    return lastUpdateInstant;
  }
  public boolean isDeleted() {
    return deleted;
  }
}

File: frontend/src/app/modules/projects/model/Employee.ts

export interface Employee {

    id: number;
    name: string;
    surname:string;
}

export class EmployeeDeserializer {

    public static fromEmployeeSimple = (value: any) : Employee => (
        {
            id: value.id,
            name: value.name,
            surname: value.surname
        }
    )
}